src/GNDStk/Tree/test/root.test.cpp

```diff
@@ -0,0 +1,21 @@
+
+#include "catch.hpp"
+#include "GNDStk.hpp"
+using namespace njoy::GNDStk;
+
+SCENARIO("Testing GNDStk tree root") {
+
+   GIVEN("A tree") {
+      Tree<> tree;
+
+      WHEN("The tree is default-constructed") {
+         CHECK(tree.root == nullptr); // it has nothing in it
+      }
+
+      WHEN("A GNDS file is read into the tree") {
+         tree.read("n-008_O_016.xml");
+         CHECK(tree.root != nullptr); // it has something in it
```

Comment:
Fortunately, I did pass Logic 101 in college, so I know this. Comment same as elsewhere, about tests not existing in a vacuum. Plenty of other tests end up illustrating that read() works correctly. I could replicate other test content here, and then someone might ask why I duplicated a test. I could replicate and then change things up a bit (for example, but changing an input file); in fact I think I did that in a place or two, and may have been asked why I did nearly the same test in two places. Invariably, and in spite of the term "unit test" existing in this business, individual code constructs and tests seldom really align in a perfect 1-on-1 manner. I can certainly go into codes like this and put some additional meat (or tofu if you prefer) into them, but I'd note that I've vetted my read() (plus, invariably, write()) functionality against the entire set of several thousand GNDS files, and would suggest that they do in fact work properly. (Those stress tests aren't included in GNDStk because, obviously, we don't want to put gigs of GNDS files into our repo, and it would probably be inappropriate to do so anyway, because those files were created by someone else.)